src/firmware/shared/physics.h

```diff
@@ -8,101 +8,112 @@
-extern const float MAX_VEL[3];
-
+/**
+ * Transformation matricies to convert speeds in the
+ * two different domains commonly used by the robot
+ *
+ * @param [in] speed4 which is the listing of wheel speeds
+ * @param [out] speed3 which is a speed in x,y,rotation in
+ * robot relative coordinates
+ * @param front_wheel_angle_deg angle between each front wheel and the y axis of the robot
+ * in degrees
+ * @param back_wheel_angle_deg angle between each back wheel and the y axis of the robot
+ * in degrees
+ */
+void shared_physics_speed4ToSpeed3(const float speed4[4], float speed3[3],
+                                   float front_wheel_angle_deg,
+                                   float back_wheel_angle_deg);
+void shared_physics_speed3ToSpeed4(const float speed3[3], float speed4[4],
+                                   float front_wheel_angle_deg,
+                                   float back_wheel_angle_deg);
 
-// transformation matricies to convert speeds in the
-// two different domains commonly used by the robot
-// speed4 which is the listing of wheel speeds
-// and speed3 which is a speed in x,y,rotation in
-// robot relative coordinates
-void speed4_to_speed3(const float speed4[4], float speed3[3]);
-void speed3_to_speed4(const float speed3[3], float speed4[4]);
+/**
+ * Transformation matricies to convert speeds in the two different domains commonly used
+ * by the robot NOTE: legacy conversion function that assumes that the angles are 55deg
+ * (front) and 45deg (back). Please use shared_physics_speed4ToSpeed3 instead
+ *
+ * @param [in] speed4 which is the listing of wheel speeds
+ * @param [out] speed3 which is a speed in x,y,rotation in
+ * robot relative coordinates
+ */
+void shared_physics_legacySpeed4ToSpeed3(const float speed4[4], float speed3[3]);
 
-// transformation matricies to convert forces in the
-// two different domains commonly used by the robot
-// force4 which is the listing of wheel forces
-// and force3 which is a force in x,y,rotation in
-// robot relative coordinates
-void force3_to_force4(float force3[3], float force4[4]);
+/**
+ * Transformation matricies to convert forces in the
+ * two different domains commonly used by the robot
+ *
+ * @param [in] force4 which is the listing of wheel forces
+ * @param [out] force3 which is a force in x,y,rotation in
+ * robot relative coordinates
+ * @param front_wheel_angle_deg angle between each front wheel and the y axis of the robot
+ * in radians
+ * @param back_wheel_angle_deg angle between each back wheel and the y axis of the robot
+ * in radians
+ */
+void shared_physics_force3ToForce4(float force3[3], float force4[4],
+                                   float front_wheel_angle_deg,
+                                   float back_wheel_angle_deg);
 
-float min_angle_delta(float, float);
+/**
+ * Min angle delta between the radians of two angles
+ *
+ * @param angle1 the first angle in radians
+ * @param angle2 the second angle in radians
```

Comment:
nit; missing `[in]` for these params. Same goes for a lot of functions in this file.